### ISSUE TITLE
CI: K8sValidatedMicroscope Runs microscope

### ISSUE THREAD

**Issue by aanm** [cilium/cilium#4879]
```
/home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:376
Expected
    <*errors.errorString | 0xc4203aa490>: {
        s: "Timeout reached: running microscope processes not found",
    }
to be nil
/home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/k8sT/microscope.go:69
```

https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/5048/execution/node/93/log/
[f0c5341e_K8sValidatedMicroscope_Runs_microscope.zip](https://github.com/cilium/cilium/files/2206311/f0c5341e_K8sValidatedMicroscope_Runs_microscope.zip)

It seems microscope does not start running. Unfortunately there's no logs from the microscope container so we can't get any useful information.

**Comment by eloycoto** [comment]
Due this is a PR, is the Microscope fix added in that code? We had problems this week with microscope and it is already fixed: https://github.com/cilium/cilium/pull/4856

**Comment by aanm** [comment]
@eloycoto that PR was rebased with master today

**Comment by ianvernon** [comment]
Related recent failure: https://jenkins.cilium.io/job/cilium-ginkgo/job/cilium/job/master/1273/testReport/k8s-1/11/K8sValidatedMicroscope_Runs_microscope/

Logs: 
[f0c5341e_K8sValidatedMicroscope_Runs_microscope.zip](https://github.com/cilium/cilium/files/2221053/f0c5341e_K8sValidatedMicroscope_Runs_microscope.zip)

**Comment by ianvernon** [comment]
@nebril can we close since #4898 was merged?

**Comment by ianvernon** [comment]
Another flake even after #4898 was merged :( 

Build link: https://jenkins.cilium.io/job/Ginkgo-CI-Tests-Pipeline/3407/testReport/junit/k8s-1/11/K8sValidatedMicroscope_Runs_microscope/

Output:
```
Stacktrace

/home/jenkins/workspace/Ginkgo-CI-Tests-Pipeline/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:376
Expected
    <*errors.errorString | 0xc420304e30>: {
        s: "Timeout reached: running microscope processes not found",
    }
to be nil
/home/jenkins/workspace/Ginkgo-CI-Tests-Pipeline/src/github.com/cilium/cilium/test/k8sT/microscope.go:69
Standard Output

Cilium pods: [cilium-nmvrw cilium-xxl76]
Netpols loaded: 
CiliumNetworkPolicies loaded: 
Endpoint Policy Enforcement:
     prometheus-core-6546477dc8-j59h6   =>   none
     cilium-health-k8s1                 =>   none
     cilium-health-k8s2                 =>   none
     coredns-78fcdf6894-kqqwq           =>   none
     coredns-78fcdf6894-mq7kp           =>   none
     microscope                         =>   none
Cilium agent "cilium-nmvrw": Status: Ok  Health: Ok Nodes "k8s1 k8s2" ContinerRuntime: Ok Kubernetes: Ok KVstore: Ok Controllers: Total 15 Failed 0
Cilium agent "cilium-xxl76": Status: Ok  Health: Ok Nodes "k8s1 k8s2" ContinerRuntime: Ok Kubernetes: Ok KVstore: Ok Controllers: Total 39 Failed 0

Standard Error

=== Test Finished at 2018-07-24T04:24:28Z====
===================== TEST FAILED =====================
cmd: kubectl get pods -o wide --all-namespaces
NAMESPACE     NAME                               READY     STATUS    RESTARTS   AGE       IP              NODE
kube-system   cilium-nmvrw                       1/1       Running   0          6m        192.168.36.11   k8s1
kube-system   cilium-xxl76                       1/1       Running   0          6m        192.168.36.12   k8s2
kube-system   coredns-78fcdf6894-kqqwq           1/1       Running   0          14m       10.10.1.243     k8s2
kube-system   coredns-78fcdf6894-mq7kp           1/1       Running   0          14m       10.10.1.166     k8s2
kube-system   etcd-k8s1                          1/1       Running   0          13m       192.168.36.11   k8s1
kube-system   kube-apiserver-k8s1                1/1       Running   0          13m       192.168.36.11   k8s1
kube-system   kube-controller-manager-k8s1       1/1       Running   0          13m       192.168.36.11   k8s1
kube-system   kube-proxy-k8r6n                   1/1       Running   0          14m       192.168.36.11   k8s1
kube-system   kube-proxy-pvpvj                   1/1       Running   0          6m        192.168.36.12   k8s2
kube-system   kube-scheduler-k8s1                1/1       Running   0          13m       192.168.36.11   k8s1
kube-system   microscope                         1/1       Running   0          2m        10.10.1.96      k8s2
prometheus    prometheus-core-6546477dc8-j59h6   1/1       Running   0          6m        10.10.1.47      k8s2

cmd: kubectl exec -n kube-system cilium-nmvrw -- cilium endpoint list
ENDPOINT   POLICY (ingress)   POLICY (egress)   IDENTITY   LABELS (source:key[=value])   IPv6                 IPv4          STATUS   
           ENFORCEMENT        ENFORCEMENT                                                                                   
37836      Disabled           Disabled          4          reserved:health               f00d::a0a:0:0:93cc   10.10.0.216   ready   

cmd: kubectl exec -n kube-system cilium-xxl76 -- cilium endpoint list
ENDPOINT   POLICY (ingress)   POLICY (egress)   IDENTITY   LABELS (source:key[=value])                              IPv6                   IPv4          STATUS   
           ENFORCEMENT        ENFORCEMENT                                                                                                                
14886      Disabled           Disabled          12453      k8s:io.cilium.k8s.policy.serviceaccount=microscope       f00d::a0a:100:0:3a26   10.10.1.96    ready   
                                                           k8s:io.kubernetes.pod.namespace=kube-system                                                           
                                                           k8s:k8s-app=microscope                                                                                
26413      Disabled           Disabled          4          reserved:health                                          f00d::a0a:100:0:672d   10.10.1.137   ready   
38442      Disabled           Disabled          64189      k8s:io.cilium.k8s.policy.serviceaccount=coredns          f00d::a0a:100:0:962a   10.10.1.243   ready   
                                                           k8s:io.kubernetes.pod.namespace=kube-system                                                           
                                                           k8s:k8s-app=kube-dns                                                                                  
48247      Disabled           Disabled          64189      k8s:io.cilium.k8s.policy.serviceaccount=coredns          f00d::a0a:100:0:bc77   10.10.1.166   ready   
                                                           k8s:io.kubernetes.pod.namespace=kube-system                                                           
                                                           k8s:k8s-app=kube-dns                                                                                  
65044      Disabled           Disabled          45586      k8s:app=prometheus                                       f00d::a0a:100:0:fe14   10.10.1.47    ready   
                                                           k8s:component=core                                                                                    
                                                           k8s:io.cilium.k8s.policy.serviceaccount=prometheus-k8s                                                
                                                           k8s:io.kubernetes.pod.namespace=prometheus                                                            
```

Logs: 
[9acb0442_K8sValidatedMicroscope_Runs_microscope.zip](https://github.com/cilium/cilium/files/2222167/9acb0442_K8sValidatedMicroscope_Runs_microscope.zip)

**Comment by eloycoto** [comment]
Trying microscope manually and I found the following: 

```
/usr/src/microscope # microscope | ts
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/local/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/bin/microscope/__main__.py", line 153, in <module>
  File "/bin/microscope/__main__.py", line 138, in main
  File "/bin/microscope/microscope/monitor/runner.py", line 98, in run
  File "/bin/microscope/microscope/monitor/runner.py", line 205, in retrieve_endpoint_data
  File "/bin/microscope/microscope/monitor/runner.py", line 205, in <listcomp>
KeyError: 'status'
Jul 24 14:33:34
Jul 24 14:33:34 closing
/usr/src/microscope #
```

**Comment by nebril** [comment]
Fixed by #4976
